packages/aws-cdk/lib/cli.ts

```diff
@@ -440,32 +440,19 @@ export async function exec(args: string[], synthesizer?: Synthesizer): Promise<n
     // Do PSAs here
     await version.displayVersionMessage();
 
-    if (shouldDisplayNotices()) {
-      if (cmd === 'notices' && argv.unacknowledged === true) {
-        await displayNotices({
-          outdir: configuration.settings.get(['output']) ?? 'cdk.out',
-          acknowledgedIssueNumbers: configuration.context.get('acknowledged-issue-numbers') ?? [],
-          ignoreCache: true,
-          unacknowledged: true,
-        });
-      } else if (cmd === 'notices') {
-        await displayNotices({
-          outdir: configuration.settings.get(['output']) ?? 'cdk.out',
-          acknowledgedIssueNumbers: [],
-          ignoreCache: true,
-        });
+    if (notices.shouldDisplay()) {
```

Comment:
[minor] We can achieve better encapsulation if we don't expose this. If notices should not be displayed, then `refresh` and `display` could be no-ops. And it would be safe to call them any time, and they would do the right thing.